data_model/src/events/pipeline.rs

```diff
@@ -29,48 +29,59 @@ pub use crate::transaction::RejectionReason as PipelineRejectionReason;
     Deserialize,
 )]
 pub struct EventFilter {
-    /// Filter by Entity if `Some`, if `None` all entities are accepted.
-    pub entity: Option<EntityType>,
-    /// Filter by Hash if `Some`, if `None` all hashes are accepted.
+    /// If `Some::<EntityKind>` filters by the [`EntityKind`]. If `None` accepts all the [`EntityKind`].
+    pub entity_kind: Option<EntityKind>,
+    /// If `Some::<StatusKind>` filters by the [`StatusKind`]. If `None` accepts all the [`StatusKind`].
+    pub status_kind: Option<StatusKind>,
+    /// If `Some::<Hash>` filters by the [`Hash`]. If `None` accepts all the [`Hash`].
     pub hash: Option<Hash>,
 }
 
 impl EventFilter {
```

Comment:
What about to add `inline` and `const` inside this impl block as much as possible?